packages/radix-vue/src/RangeCalendar/RangeCalendarRoot.vue

```diff
@@ -234,40 +239,51 @@ const {
   focusedValue,
 })
 
-watch(modelValue, () => {
-  if (modelValue.value.start && modelValue.value.end) {
-    if (startValue.value && modelValue.value.start.compare(startValue.value) !== 0)
-      startValue.value = modelValue.value.start.copy()
+watch(modelValue, (modelValue) => {
+  if (modelValue.start && modelValue.end) {
+    if (startValue.value && !isEqualDay(startValue.value, modelValue.start))
+      startValue.value = modelValue.start.copy()
 
-    if (endValue.value && modelValue.value.end.compare(endValue.value) !== 0)
-      endValue.value = modelValue.value.end.copy()
+    if (endValue.value && !isEqualDay(endValue.value, modelValue.end))
+      endValue.value = modelValue.end.copy()
   }
 })
 
 watch(startValue, (value) => {
   if (value && !isEqualDay(value, placeholder.value))
     onPlaceholderChange(value)
+
+  emits('update:startValue', value)
 })
 
-watch([startValue, endValue], () => {
-  if (modelValue.value && modelValue.value.start && modelValue.value.end && startValue.value && endValue.value && isSameDay(modelValue.value.start, startValue.value) && isSameDay(modelValue.value.end, endValue.value))
+watch([startValue, endValue], ([startValue, endValue]) => {
```

Comment:
let's rename the callback payload `[startValue, endValue]` to something different ya. Don't want it to be confuse with the reactive version of that. Maybe `[_startValue, _endValue]` ?

Reply:
Sure thing, I'll update the code and add a story in `Chromatic`